x/ccv/provider/keeper/relay.go

```diff
@@ -144,21 +196,35 @@ func (k Keeper) QueueVSCPackets(ctx sdk.Context) {
 	}
 
 	for _, chainID := range k.GetAllRegisteredConsumerChainIDs(ctx) {
-		currentValidators := k.GetConsumerValSet(ctx, chainID)
+		currentValidators, err := k.GetConsumerValSet(ctx, chainID)
+		if err != nil {
+			panic(fmt.Errorf("failed to get consumer validators: %w", err))
+		}
 		topN, _ := k.GetTopN(ctx, chainID)
 
 		if topN > 0 {
 			// in a Top-N chain, we automatically opt in all validators that belong to the top N
-			minPower, err := k.ComputeMinPowerInTopN(ctx, bondedValidators, topN)
-			if err == nil {
-				// set the minimal power of validators in the top N in the store
-				k.SetMinimumPowerInTopN(ctx, chainID, minPower)
-
-				k.OptInTopNValidators(ctx, chainID, bondedValidators, minPower)
-			} else {
+			// of the active validators
+			activeValidators, err := k.GetLastActiveBondedValidators(ctx)
+			var minPower int64
+			if err != nil {
 				// we just log here and do not panic because panic-ing would halt the provider chain
-				k.Logger(ctx).Error("failed to compute min power to opt in for chain", "chain", chainID, "error", err)
+				k.Logger(ctx).Error("failed to get active validators", "error", err)
+				// assume that the minPower is 0, since we cannot compute it because the validator set is broken
```

Comment:
This error handling is not consistent. In some cases we panic (e.g.,  ProviderValidatorUpdates), while in other we just log an error and continue. We need to make this consistent. and add comments on the design decisions. 

In this case, why would `GetLastActiveBondedValidators` return an error? If it's because the state is corrupted, then we should panic. Otherwise, we need to ask if it's safe to continue. Assuming minPower is zero means that all validators need to run TopN chains, which will lead to a bunch of small validators getting jailed for downtime as they don't expect that.

Reply:
Changed this to panic. It's what the sdk does when the underlying method fails (IterateLastValidatorPowers)